File: src/main/java/com/vspk/bookrest/repository/VerificationRepository.java
package com.vspk.bookrest.repository;

import com.vspk.bookrest.domain.Verification;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface VerificationRepository extends JpaRepository<Verification, Long> {
    Optional<Verification> findVerificationByCode(String code);
}

File: src/main/java/com/vspk/bookrest/repository/VerificationRepository.kt
package com.vspk.bookrest.repository

import com.vspk.bookrest.domain.Verification
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface VerificationRepository : JpaRepository<Verification, Long> {
    fun findVerificationByCode(code: String): Optional<Verification>
}